raimitigations/cohort/cohort_handler.py

```diff
@@ -0,0 +1,538 @@
+        prev_cohort_def = []
+        for i, cohort_def in enumerate(self.cohort_def):
+            if cohort_def is None and i < len(self.cohort_def) - 1:
+                raise ValueError("ERROR: only the last cohort is allowed to have a condition list assigned to 'None'.")
+            cohort = self._instantiate_cohort(cohort_def, self._cohort_names[i])
+            if cohort_def is None:
+                cohort.create_query_remaining_instances_cohort(prev_cohort_def)
+            else:
+                prev_cohort_def.append(cohort_def)
+            self.cohorts.append(cohort)
+
+    # -----------------------------------
+    def _check_intersection_cohorts(self, index_used: list):
+        """
+        Checks if there are any index that repeats in the ``index_used`` list.
+        If that happens, this means that at least one instance of the dataset
+        belongs to more than one cohort, which is not allowed.
+
+        :param index_used: a list with the index of all instances that have a
+            mathcing cohort.
+        """
+        set_index = list(set(index_used))
+        if len(set_index) != len(index_used):
+            raise ValueError("ERROR: some rows of the dataset belong to more than one cohort.")
+
+    # -----------------------------------
+    def _merge_cohort_datasets(self, cht_df: dict, org_index: list = None):
+        """
+        Merges all cohort subsets (after going through their own
+        transformations) into a single dataset. After concatanating all
+        predictions into a single array, it is reordered so that it keeps
+        the same order as the original dataset. The ``org_index`` is
+        is used to determine this order.
+
+        :param cht_df: a dictionary where each key is a cohort
+            name, and the value associated to each key is a dataframe
+            representing the subset that belongs to that cohort;
+        :param org_index: the index list of the original dataset. This
+            is used to define the order that the final predictions should
+            be sorted so that it uses the same instance order used in the
+            original dataset;
+        :return: a dataset containing the subset of cohorts after their
+            transformations. If ``org_index`` is provided, the resulting
+            dataset will follow the same order as the original dataset;
+        :rtype: pd.DataFrame
+        """
+        if not self._cohorts_compatible:
+            self.print_message(
+                "WARNING: the transformations used over the cohorts resulted in each cohort having different "
+                + "columns. The transform() method will return a list of transformed subsets (one for each cohort)."
+            )
+            return cht_df
+
+        final_df = None
+        for df in cht_df.values():
+            if final_df is None:
+                final_df = df
+            else:
+                final_df = pd.concat([final_df, df], axis=0)
```

Comment:
you don't really need to do this if, pd.concat([df,None]) = df, you can just do:
```
final_df = None
for df in cht.values()
   final_df = pd.concat([final_df, df] axis=0)
```

Reply:
That's a lot easier, that's for sure! I've done this "_if final_df is None..._" structure so many times.....if only I knew this earlier. Thanks a lot for letting me know!